Patch for karchive to fix failed compilation

### Describe the bug

[static_cast_mode.patch](https://github.com/user-attachments/files/23890890/static_cast_mode.patch)

When compiling this on Gentoo I've been running into this compilation issue:

```
/usr/include/qt6/QtCore/qstring.h:364:5: note: template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h: In substitution of ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const [with Args = {QFlags<QIODeviceBase::OpenModeFlag>&}]’:
/var/tmp/portage/dev-games/tiled-1.11.2_p20250614/work/tiled-f0191439b7f6b18cd727ee051e976756889b4d11/src/karchive/src/karchive.cpp:203:53:   required from here
  203 |         setErrorString(tr("Unsupported mode %1").arg(d->mode));
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/usr/include/qt6/QtCore/qstring.h:364:5: error: no type named ‘type’ in ‘struct std::enable_if<false, QString>’
  364 |     arg(Args &&...args) const
      |     ^~~
```

I'm not super familiar with QT, but I fixed it by calling `static_cast<int>()` on uses of mode, maybe it's an enum and needs an int?

My system has: QTCore 5.15.18
Tiled version was: 1.11.2_p20250614

I'll attach the patch I made to allow it to compile on my system

Patch attached, hope this helps somehow.


### Steps to reproduce

emerge it on gentoo.

### Expected behavior

no compile errors.

### OS version

Gentoo Linux

### Tiled version

1.11.2_p20250614

--- a/src/karchive/src/kar.cpp
+++ b/src/karchive/src/kar.cpp
@@ -90,7 +90,7 @@ bool KAr::openArchive(QIODevice::OpenMode mode)
         return true;
     }
     if (mode != QIODevice::ReadOnly && mode != QIODevice::ReadWrite) {
-        setErrorString(tr("Unsupported mode %1").arg(mode));
+        setErrorString(tr("Unsupported mode %1").arg(static_cast<int>(mode)));
         return false;
     }
 

--- a/src/karchive/src/karchive.cpp
+++ b/src/karchive/src/karchive.cpp
@@ -152,7 +152,7 @@ bool KArchive::open(QIODevice::OpenMode mode)
     }
 
     if (!d->dev->isOpen() && !d->dev->open(mode)) {
-        setErrorString(tr("Could not set device mode to %1").arg(mode));
+        setErrorString(tr("Could not set device mode to %1").arg(static_cast<int>(mode)));
         return false;
     }
 
@@ -200,7 +200,7 @@ bool KArchive::createDevice(QIODevice::OpenMode mode)
         }
         break; // continued below
     default:
-        setErrorString(tr("Unsupported mode %1").arg(d->mode));
+        setErrorString(tr("Unsupported mode %1").arg(static_cast<int>(d->mode)));
         return false;
     }
     return true;

--- a/src/karchive/src/krcc.cpp
+++ b/src/karchive/src/krcc.cpp
@@ -110,7 +110,7 @@ bool KRcc::openArchive(QIODevice::OpenMode mode)
         return true;
     }
     if (mode != QIODevice::ReadOnly && mode != QIODevice::ReadWrite) {
-        setErrorString(tr("Unsupported mode %1").arg(mode));
+        setErrorString(tr("Unsupported mode %1").arg(static_cast<int>(mode)));
         return false;
     }
 

